lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -991,6 +996,276 @@ def _generate_report(self, trans, invocation_id, **kwd):
+                        if hasattr(job_input.dataset, 'dataset_id'):
+                            encoded_dataset_id = trans.security.encode_id(job_input.dataset.dataset_id)
+                            input_obj = {
+                                'filename': job_input.dataset.name,
+                                'uri': url_for('history_content', history_id=encoded_history_id, id=encoded_dataset_id, qualified=True),
+                                'access_time': job_input.dataset.create_time.isoformat(),
+                            }
+                            input_list.append(input_obj)
+
+                    for job_output in job.output_datasets:
+                        if hasattr(job_output.dataset, 'dataset_id'):
+                            encoded_dataset_id = trans.security.encode_id(job_output.dataset.dataset_id)
+                            output_obj = {
+                                'filename': job_output.dataset.name,
+                                'uri': url_for('history_content', history_id=encoded_history_id, id=encoded_dataset_id, qualified=True),
+                                'access_time': job_output.dataset.create_time.isoformat(),
+                            }
+                            output_list.append(output_obj)
+
+                        if job_output.name in workflow_outputs_list:
+                            output = {
+                                'mediatype': job_output.dataset.extension,
+                                'uri': {
+                                    'filename': job_output.dataset.name,
+                                    'uri': url_for('history_content', history_id=encoded_history_id, id=encoded_dataset_id, qualified=True),
+                                    'access_time': job_output.dataset.create_time.isoformat(),
+                                }
+                            }
+                            output_subdomain.append(output)
+                current_tool = dict_workflow['steps'][str(i)]
+                pipeline_step = {
+                    'step_number': i,
+                    'name': current_tool['name'],
+                    'description': current_tool['annotation'],
+                    'version': current_tool['tool_version'],
+                    'prerequisite': kwd.get('prerequisite', []),
+                    'input_list': input_list,
+                    'output_list': output_list
+                }
+                pipeline_steps.append(pipeline_step)
+                try:
+                    software_prerequisite = {
+                        'name': current_tool['tool_shed_repository']['name'],
+                        'version': current_tool['tool_version'],
+                        'uri': {
+                            'uri': current_tool['content_id'],
+                            'access_time': current_tool['uuid']
+                        }
+                    }
+                    if software_prerequisite['uri']['uri'] not in tools:
```

Comment:
This is a mix of the Galaxy tool id for tool steps and a part of the tool shed repository (the name). This combination is ... odd. What exactly do you want to capture here ?

Reply:
See [this](https://github.com/biocompute-objects/BCO_Specification/blob/main/docs/execution-domain.md#253-algorithmic-tools-and-software-prerequisites-software_prerequisites) for an explantation.